File: Java/JAVA 3.0/RECURSION/ARRAYS/sum.java
package RECURSION.ARRAYS;

public class sum {
    static int sum(int[] arr,int idx)
    {
        if(idx == arr.length-1)
        {
            return arr[idx];
        }
        return  arr[idx]+sum(arr,idx+1);
    }

    public static void main(String[] args) {
        int[] arr = {2,3,5,20,1};
        System.out.println("Sum of the array is: "+sum(arr,0));
    }
}

File: C/recursion.c/sum.c
// print sum of 1st n natural numbers using recursion
#include<stdio.h>
int sum(int n);
void main(){
    int num;
    printf("Enter any number\n");
    scanf("%d",&num);
    printf("%d",sum(num));
}
int sum(int n){
    if(n>0){
        return n + sum(n-1);
    }
}
